cloudflared/DOCS.md

```diff
@@ -144,6 +148,41 @@ additional_hosts:
 anymore. Nevertheless, you should also manually delete the DNS entry from
 Cloudflare since it can not be deleted by the Add-on._
 
+### Option: `strict_ssl`
+
+When enabled, the tunnel in Cloudflare will be set with the [_Full (strict)_ SSL/TLS encryption mode](https://developers.cloudflare.com/ssl/origin-configuration/ssl-modes#full-strict) to provide a higher security.
+
+In this mode, the local connection needs to be stablished over HTTPS, presenting a certificate issued by a publicly trusted certificate authority (like using the [Let's Encrypt addon](https://github.com/home-assistant/addons/tree/master/letsencrypt#home-assistant-add-on-letsencrypt)), or a [Cloudflare Origin CA](https://blog.cloudflare.com/cloudflare-ca-encryption-origin/) one (which [the addon can generate for you automatically](#option-generatecert)).
+
+Here is an example on how to configure the Home Assistant HTTP server to use HTTPS with the appropriate certificate:
+
+```diff-json
+// /config/configuration.yaml
+
+http:
+  use_x_forwarded_for: true
+  trusted_proxies:
+    - 172.30.33.0/24
++ ssl_certificate: /ssl/fullchain.pem
++ ssl_key: /ssl/privkey.pem
+```
+
+### Option: `generate_cert`
+
+When enabled, the addon will generate, retrieve and export a Cloudflare Origin CA issued certificate for you in the `/ssl` folder.
```

Comment:
Should we generate a new origin certificate or just extract the origin zone certificate from the already generated cert.pem file (after cloudflared login)? 
I would suggest to extract the cert and key from cert.pem, otherwise we would need to pas another Cloudflare API token (afaik) to the add-on.

Reply:
I guess we have to get an new certificate from Cloudflare, with a long enough validity for this to actually make sense. I think we can use the certificate for the communication with the Cloudflare API though. Did not really look into it to be honest, so if it is working with extracting everything from what we have already, happy to do so

Reply:
I did not dig into it already, so I don't know how the authentication with the Cloudflare API will be made yet. I will share more details once I start working on it.